### PR TITLE
Implement multiple clipboards

### LINKED ISSUE #298: Implement support for multiple clipboards
Currently we use the main clipboard for moving layers. The `CopySelectedLayers` and `PasteLayers` Messages should take the clipboard as an argument with the clipboard being an enum used to index an array of clipboards.
Relevant files: `document_message_handler.rs` `input_mapper.rs`

### PR DESCRIPTION
**Pull request by TrueDoctor**
Closes #298

### CODE REVIEW

File: editor/src/document/document_message_handler.rs [hunk 1]
```diff
@@ -10,15 +10,24 @@ use std::collections::{HashMap, VecDeque};
 use super::DocumentMessageHandler;
 use crate::consts::DEFAULT_DOCUMENT_NAME;
 
+#[derive(Serialize, Deserialize, Clone, Copy, PartialEq, Eq, Debug)]
+pub enum Clipboard {
+	System,
+	User,
+	_ClipbordCount,
+}
+static CLIPBORD_COUNT: u8 = Clipboard::_ClipbordCount as u8;
```
Comment (Keavon) [inline editor/src/document/document_message_handler.rs:19]: Typo fix: `CLIPBOARD_COUNT`

File: editor/src/document/document_message_handler.rs [hunk 2]
```diff
@@ -10,15 +10,24 @@ use std::collections::{HashMap, VecDeque};
 use super::DocumentMessageHandler;
 use crate::consts::DEFAULT_DOCUMENT_NAME;
 
+#[derive(Serialize, Deserialize, Clone, Copy, PartialEq, Eq, Debug)]
+pub enum Clipboard {
+	System,
+	User,
+	_ClipbordCount,
```
Comment (Keavon) [inline editor/src/document/document_message_handler.rs:17]: Typo fix: `_ClipboardCount`

File: editor/src/document/document_message_handler.rs [hunk 3]
```diff
@@ -267,19 +276,19 @@ impl MessageHandler<DocumentsMessage, &InputPreprocessor> for DocumentsMessageHa
...
 			}
-			Copy => {
+			Copy(clipboard) => {
 				let paths = self.active_document().selected_layers_sorted();
 				self.copy_buffer.clear();
 				for path in paths {
 					match self.active_document().graphene_document.layer(&path).map(|t| t.clone()) {
 						Ok(layer) => {
-							self.copy_buffer.push(layer);
+							self.copy_buffer[clipboard as usize].push(layer);
```
Comment (Keavon) [inline editor/src/document/document_message_handler.rs:285]: How is the `clipboard` enum able to be cast to a `usize`?

File: editor/src/document/document_message_handler.rs [hunk 4]
```diff
@@ -307,11 +317,11 @@ impl MessageHandler<DocumentsMessage, &InputPreprocessor> for DocumentsMessageHa
 					)
 				};
 				if insert_index == -1 {
-					for layer in self.copy_buffer.iter() {
+					for layer in self.copy_buffer[clipboard as usize].iter() {
 						paste(layer, responses)
 					}
 				} else {
-					for layer in self.copy_buffer.iter().rev() {
+					for layer in self.copy_buffer[clipboard as usize].iter().rev() {
```
Comment (Keavon) [inline editor/src/document/document_message_handler.rs:324]: Same question here.

File: editor/src/input/input_mapper.rs [hunk 5]
```diff
@@ -133,7 +134,7 @@ impl Default for Mapping {
 		// it as an available action in the respective message handler file (such as the bottom of `document_message_handler.rs`)
 		let mappings = mapping![
 			// Higher priority than entries in sections below
-			entry! {action=DocumentsMessage::Paste, key_down=KeyV, modifiers=[KeyControl]},
+			entry! {action=DocumentsMessage::Paste(User), key_down=KeyV, modifiers=[KeyControl]},
```
Comment (Keavon) [inline editor/src/input/input_mapper.rs:137]: Where does the `User` symbol come from? I don't see it above in the file.

**Review by Keavon — APPROVED** [review]
Looks good!

File: editor/src/document/document_message_handler.rs (hunk 3 above)
Comment (TrueDoctor) [reply editor/src/document/document_message_handler.rs:285]: The enum is represented as an integer of the appropriate size which is then cast to a usize

File: editor/src/input/input_mapper.rs (hunk 5 above)
Comment (TrueDoctor) [reply editor/src/input/input_mapper.rs:137]: Thats part of `use Clipboard::*`

**Pull request merged**